aiida/backends/utils.py

```diff
@@ -68,6 +71,22 @@ def get_settings_manager():
     return manager
 
 
+def validate_schema_generation():
+    """Verify that the database schema generation is compatible with the current code schema generation."""
+    from aiida.common.exceptions import ConfigurationError
+    if get_db_schema_generation() != SCHEMA_GENERATION_VALUE:
+        raise ConfigurationError(
+            'The schema generation of your database is newer than that of the code and is therefore incompatible.'
+        )
+
+
+def get_db_schema_generation():
+    """Get the schema generation of the current database."""
+    from aiida.backends.utils import get_settings_manager
+    manager = get_settings_manager()
+    return manager.get(SCHEMA_GENERATION_KEY)
```

Comment:
Thinking better about this, I would do `manager.get(SCHEMA_GENERATION_KEY, 1)` (does this `.get` accept default values?) because otherwise this will always fail for anything before a few commits ago, when the key was added. Do you agree?

Reply:
It does support a default, but I explicitly didn't. The reason is that "a few commits before" this code doesn't exist, so it cannot except

Reply:
In fact the `get` does not have a default, but I just catch the exception and use the default